flink-tests/src/test/java/org/apache/flink/test/checkpointing/ZooKeeperHighAvailabilityITCase.java

```diff
@@ -366,6 +369,11 @@ public void snapshotState(FunctionSnapshotContext context) throws Exception {
 				failInCheckpointLatch.await();
 				if (!failedAlready.getAndSet(true)) {
 					throw new RuntimeException("Failing on purpose.");
+				} else {
```

Comment:
Instead of this else branch, couldn't we always fail unless `afterMessWithZooKeeper` is `true`?